src/table/components/PaginationContent.jsx

```diff
@@ -0,0 +1,189 @@
+  const { page, rowsPerPage, rowsPerPageOptions } = pageInfo;
+
+  if (!paginationNeeded) return null;
+
+  const onFirstPage = page === 0;
+  const onLastPage = page >= lastPageIdx;
+  const tabIndex = !keyboard.enabled || keyboard.active ? 0 : -1;
+  const width = footerContainer ? footerContainer.getBoundingClientRect().width : rect.width;
+  const showFirstAndLast = shouldShow('firstLast', width);
+  const showRowsPerPage = !selectionsAPI.isModal() && shouldShow('rppOptions', width) && totalColumnCount <= 100;
+  const displayedRowsText = translator.get('SNTable.Pagination.DisplayedRowsLabel', [
+    `${page * rowsPerPage + 1} - ${Math.min((page + 1) * rowsPerPage, totalRowCount)}`,
+    totalRowCount,
+  ]);
+
+  const handleChangeRowsPerPage = (evt) => {
+    setPageInfo({ ...pageInfo, page: 0, rowsPerPage: +evt.target.value });
+    announce({ keys: [['SNTable.Pagination.RowsPerPageChange', evt.target.value]], politeness: 'assertive' });
+  };
+
+  const handleSelectPage = (event) => handleChangePage(+event.target.value);
+
+  const handleLastButtonTab = keyboard.enabled ? (event) => handleLastTab(event, selectionsAPI.isModal()) : null;
+
+  const selectStyle = {
+    backgroundColor: 'inherit',
+    '& .MuiNativeSelect-icon': { color: theme.table.pagination.iconColor },
+  };
+
+  const getButton = (disabledCondition, pageNumber, type, onKeyDown = null) => {
+    const iconType = `${type}${direction === 'rtl' ? 'RTL' : ''}`;
+    const IconComponent = icons[iconType];
+    const buttonStyle = disabledCondition
+      ? {
+          color: theme.table.pagination.disabledIconColor,
+          cursor: 'default',
+        }
+      : { color: theme.table.pagination.iconColor };
+
+    return (
+      <IconButton
+        data-testid="pagination-action-icon-button"
+        onClick={!disabledCondition ? () => handleChangePage(pageNumber) : null}
+        aria-disabled={disabledCondition}
+        ok
+        aria-label={translator.get(`SNTable.Pagination.${type}`)}
+        title={!constraints.passive && translator.get(`SNTable.Pagination.${type}`)}
+        tabIndex={tabIndex}
+        sx={{ ...buttonStyle, height: 32 }}
+        onKeyDown={onKeyDown}
+      >
+        <IconComponent />
+      </IconButton>
+    );
+  };
+
+  const getDropdown = (name, value, options, handleChange) => {
+    const translationName = `SNTable.Pagination.${name}`;
+    const id = `${name}-dropdown`;
+    const inputProps = {
+      tabIndex,
+      id,
+      'data-testid': id,
+      style: { color: theme.table.pagination.color, height: 30 },
+    };
+
+    return (
+      <FormControl sx={{ px: 2.5 }}>
+        <InputLabel sx={{ color: theme.table.pagination.color }} htmlFor={id} shrink={false}>
+          {`${translator.get(translationName)}:`}
+        </InputLabel>
+        <Select sx={selectStyle} native value={value} onChange={handleChange} inputProps={inputProps}>
+          {options}
+        </Select>
```

Comment:
I was thinking that only rendering the option when we click the select button to improve performance.

Reply:
yes possibly. Maybe not much so can have a look at it later